lib/Sema/TypeCheckDeclPrimary.cpp

```diff
@@ -2147,6 +2147,27 @@ class DeclChecker : public DeclVisitor<DeclChecker> {
           } else {
             VD->diagnose(diag::property_does_not_override, isClassContext)
                 .highlight(OA->getLocation());
+            //If the override is a property-decl, check if there exists
+            //a no-args func-decl in the base class, if yes, provide a fix-it.
+            auto declName = VD->getBaseName();
+            auto declType = VD->getResultTypeRepr();
+            auto inherited = DC->getSelfClassDecl()->getInherited();
+            //If there is an inherited context, perform member lookup.
+            if (!inherited.empty()) {
+              auto superClass = inherited.front().getType();
+              auto lookupResult = superClass->getAnyNominal()->lookupDirect(declName);
```

Comment:
Ok, I realized that I did mislead you a bit - you actually want to use `TypeChecker::lookupMember` with `DC->getSelfClassDecl()` because the member could be located farther in the superclass chain too.

Reply:
That's fine, that led me to figuring out more things and increased my knowledge a ton! 

Although, could you give me a brief explanation for each type of lookup, i.e., the `constraints`, `lookupdirect` and `lookupmember`, according to my understanding, the constraint system performs a more generalised form of lookup and is too overkill for this context. Other two, I am not too sure about...

Reply:
`lookupDirect` finds only members directly on the given type and its extensions, `lookupMember` is going to look at all the supertypes too, there is also unqualified lookup (`TypeChecker::lookupUnqualified`) it would lookup declarations with the given name in the given declaration context.

Reply:
Ahh! Thanks for the info! I will make the changes!

Reply:
@xedin Correct me if I am wrong, I have tried the following ways to fetch the required member, but none of them give any results: 

`TypeChecker::lookupMember(DC->getSelfClassDecl(), declReturnType, DeclNameRef(declName))`
`TypeChecker::lookupMember(DC, declReturnType, DeclNameRef(declName))`

Reply:
You may realize the mistake if you read the documentation for `TypeChecker::lookupMember`.

Reply:
Thanks @AnthonyLatsis, will look into the same!

Reply:
@AnthonyLatsis 
I have tried figuring out with several different possibilities and also looked through use cases and examples across the codebase, but didn't quite figure out yet. All of them throw the null pointer error or similar.

Reply:
Need help?

Reply:
Well, not yet! I wanna try my best!

Reply:
Regarding the `TypeChecker::lookupMember`, how does it exactly lookup for the member, in the sense that, the first parameter I suppose would take the class decl, i.e., `DC->getSelfClassDecl()`, the second parameter, which takes the `Type`, would be the exact return type, as I tried above or would it be of some other form due to which it fails to fetch any results (first and third parameter I suppose I have passed correctly)?

Reply:
Sometimes it's helpful to take a look at some existing call sites i.e. TypeCheckAttr.cpp has a lot of uses. First argument is a declaration context that is requesting the member, `DC` in your case, the second argument is the type of class you are trying to lookup members for, which is `DC->getSelfClassDecl()->getDeclaredType()`.

Reply:
https://github.com/apple/swift/blob/ad3fec4e7e73be7ad077fb66e9ce3edbc2fe7240/lib/Sema/TypeChecker.h#L886-L896

Reply:
Thanks for helping out! @xedin @AnthonyLatsis!

Reply:
> https://github.com/apple/swift/blob/ad3fec4e7e73be7ad077fb66e9ce3edbc2fe7240/lib/Sema/TypeChecker.h#L886-L896
> 
> 
> 
> `type` is not the return type of the members it should look for.

I had realised that pretty early, cause initially, yes, I was passing the function return type, but was stuck after that, thanks!

Reply:
`Assertion failed: (Val && "isa<> used on a null pointer")`

Reply:
`Assertion failed: (Val && "isa<> used on a null pointer"), function doit, file Casting.h, line 104.`

Still the same error.

Reply:
Seems to be a problem with this line, `genericParams = cast<ExtensionDecl>(this)->getGenericParams();`

Reply:
Looks like you'd actually have to get this under the debugger to see what is going on, I don't know why lookup would be crashing like this.

Reply:
Hmm.

Reply:
@xedin 
Hey! I managed to debug it!

So it is a function parameter-cased issue. Essentially, when there aren't any parameters in the function, it crashes!

Reply:
Ignore the last deleted comment, I hadn't changed the if else part.

Reply:
Meanwhile, debugging to what could be the cause.

Reply:
FYI that `DC->getSelfClassDecl()->getSuperclass()` could produce an `Type()` which is effectively a `nullptr`.

Reply:
Well, but, when I did `llvm::errs() << DC->getSelfClassDecl()->getSuperclass();`, it does show class 'C' which in my case is the super class.

Reply:
That's fine, I'm just saying that it could be.